python/lsst/daf/butler/remote_butler/_remote_butler.py

```diff
@@ -188,15 +153,19 @@ def isWriteable(self) -> bool:
     @property
     def dimensions(self) -> DimensionUniverse:
         # Docstring inherited.
-        if self._dimensions is not None:
-            return self._dimensions
+        with self._cache.access() as cache:
```

Comment:
Does this need the read lock? If the value `is not None` then no other thread is going to make it `None`.

Reply:
I have not been able to locate any definitive information about Python's memory model for threading other than "if you use the synchronization primitives from the threading module then it works how you would expect".  

We probably don't need this on the read in the current version of CPython because of the GIL but that's not documented as permanent behavior anywhere that I am aware of.